From 74e6f0539328a6c4cd3d9e23a9a61ce540b6bd00 Mon Sep 17 00:00:00 2001
From: Yaroslav Lavrinenko <y.lavrinenko@samsung.com>
Date: Mon, 3 Apr 2023 15:59:42 +0300
Subject: [PATCH 089/225] [dba] Set the DBA's allocator granularity to 2M

Set the allocator granularity to 2M to fit the DAX restrictions.

Resolve: MCS23-764
---
 drivers/dba_resource/dba_allocator.c | 2 +-
 include/uapi/linux/dba_resources.h   | 3 +++
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/dba_resource/dba_allocator.c b/drivers/dba_resource/dba_allocator.c
index 9e79e5e61..d49388cea 100644
--- a/drivers/dba_resource/dba_allocator.c
+++ b/drivers/dba_resource/dba_allocator.c
@@ -20,7 +20,7 @@ DEFINE_MUTEX(allocator_lock);
 
 static int init_memory_pool(struct MemoryAllocator *alloc)
 {
-	alloc->granularity = PAGE_SIZE;
+	alloc->granularity = DBA_MEMORY_ADDRESS_ALIGN;
 
 	alloc->pool = gen_pool_create(ilog2(alloc->granularity), -1);
 
diff --git a/include/uapi/linux/dba_resources.h b/include/uapi/linux/dba_resources.h
index 2caa8c17e..dadc0d1ac 100644
--- a/include/uapi/linux/dba_resources.h
+++ b/include/uapi/linux/dba_resources.h
@@ -25,6 +25,9 @@
 #define DBA_MEMORY_TOTAL_SIZE (4ULL * 1024 * 1024 * 1024)
 #define DBA_RW_MODE 0666
 
+// The default alignment for DAX device is 2M
+#define DBA_MEMORY_ADDRESS_ALIGN 0x200000
+
 struct dba_allocation {
 	uint64_t address;
 	uint64_t size;
-- 
2.34.1

